src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1381,6 +1381,20 @@
   (let [last-modified-milliseconds (.lastModified (io/file cacrl))]
        (time-coerce/from-long last-modified-milliseconds)))
 
+(schema/defn ^:always-validate update-crl :- CertificateRevocationList
+  [upstream-crl-pem :- InputStream
+   crl-path :- schema/Str]
+  (log/info (i18n/trs "Updating CRL at {0}" crl-path))
+  (let [upstream-crls (utils/pem->crls upstream-crl-pem)
+        our-crl (first (utils/pem->crls crl-path))
+        new-crl-chain (cons our-crl upstream-crls)]
+    ;; what verification do we need?
```

Comment:
I think in Puppet's Ruby code we used to create an SSLContext like object and validate that it is well formed. We might be able to do something similar here, load the ca certs and validate that the existing CA cert is still valid given the new CRLs